### PR TITLE
Add networking testmod

### PR DESCRIPTION
**Pull request by BasiqueEvangelist**

### CODE REVIEW

**Review by yitzy299 — CHANGES_REQUESTED** [review]
I think it would be best here to create a new module as opposed to adding to the main package, or even a seperate mod itself might be the way forward

File: testmod/src/main/java/us/potatoboy/ledger/testmod/mixin/ClientPlayerInteractionManagerMixin.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package us.potatoboy.ledger.testmod.mixin;
```
Comment (yitzy299) [inline testmod/src/main/java/us/potatoboy/ledger/testmod/mixin/ClientPlayerInteractionManagerMixin.java:1]: Should be in src/testmod/java/us/potatoboy/ledger/testmod/mixin

**Review by yitzy299 — CHANGES_REQUESTED** [review]
Looks relatively good - one change and then I'll start testing

File: testmod/build.gradle [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    relocate "com.moandjiezana.toml", relocPath + "com.moandjiezana.toml"
+    relocate "com.uchuhimo.konf", relocPath + "com.uchuhimo.konf"
+    relocate "javassist", relocPath + "javassist"
+    // Relocate each apache lib separately as just org.apache.commons will relocate things that aren't shaded and break stuff
+    relocate "org.apache.commons.lang3", relocPath + "org.apache.commons.lang3"
+    relocate "org.apache.commons.text", relocPath + "org.apache.commons.text"
+    relocate "org.reflections", relocPath + "org.reflections"
+    // it appears you cannot relocate sqlite due to the native libraries
+//    relocate "org.sqlite", relocPath + "org.sqlite"
+}
```
Comment (Tom-The-Geek) [inline testmod/build.gradle:112]: Is there a reason all of the shadow stuff is repeated in the subproject buildscript, it doesn't really seem like its needed.

**Review by Tom-The-Geek — CHANGES_REQUESTED** [review]
Looks good apart from the repeated shadow stuff that doesn't look needed

File: testmod/src/main/java/us/potatoboy/ledger/testmod/mixin/ClientPlayerInteractionManagerMixin.java (hunk 1 above)
Comment (BasiqueEvangelist) [reply testmod/src/main/java/us/potatoboy/ledger/testmod/mixin/ClientPlayerInteractionManagerMixin.java:1]: Huh?
Do you mean I should put it in the `testmod` source set of the `testmod` subproject?

File: testmod/build.gradle (hunk 2 above)
Comment (BasiqueEvangelist) [reply testmod/build.gradle]: As it turns out, the project builds (and runs in a production environment) without shadowing anything, so I just removed the shadow plugin.

**Review by Tom-The-Geek — DISMISSED** [review]
Looks good, just need clarification from @yitzy299 on what he meant about the location of the files

**Review by yitzy299 — APPROVED** [review]
Looks good

**Review by Tom-The-Geek — APPROVED** [review]

**Pull request merged**